core-rust/state-manager/src/state_manager.rs

```diff
@@ -1115,3 +1090,55 @@ struct PendingTransactionResult {
     pub invalid_at_epoch: Epoch,
     pub rejection_reason: Option<RejectionReason>,
 }
+
+/// An internal implementation delegate dealing with intent hash duplicates.
+// TODO: Remove after this responsibility is implemented by the Engine.
+struct DuplicateIntentHashDetector<'s, S> {
+    store: &'s S,
+    recorded_intent_hashes: NonIterMap<IntentHash, IntentHashDuplicateWith>,
```

Comment:
Super minor, but we could move `IntentHashDuplicateWith` down here to be alongside the definition of `DuplicateIntentHashDetector`